src/expr/src/vector_op/cast.rs

```diff
@@ -376,18 +377,23 @@ pub fn int32_to_bool(input: i32) -> Result<bool> {
 
 // For most of the types, cast them to varchar is similar to return their text format.
 // So we use this function to cast type to varchar.
-pub fn general_to_text<T: ToText>(elem: T) -> Result<Box<str>> {
-    Ok(elem.to_text().into_boxed_str())
+pub fn general_to_text(elem: impl ToText, writer: &mut dyn Write) -> Result<()> {
```

Comment:
I'm not sure whether `dyn` is good here. 🤔 Seems we're sure that this is the `PartialStringWriter` type.

Reply:
Another type used is `String`. I tried `&mut impl Write` but failed. 😭

Reply:
If there're two, we may use a type alias with `Either`? 🤣

Reply:
Then seems we don't need this dynamic wrapper anymore. 🤔

Reply:
That looks really weird. I'd rather using dynamic. 🥵

Reply:
Actually, the performance of `eval_row` does not that matter. 🥵 Both LGTM.